python/lsst/daf/butler/cli/cmd/commands.py

```diff
@@ -269,3 +268,24 @@ def query_data_ids(**kwargs):
             print("No results. Try requesting some dimensions or datasets, see --help for more information.")
         else:
             print("No results. Try --help for more information.")
+
+
+@click.command(cls=ButlerCommand)
+@repo_argument(required=True)
+@element_argument(required=True)
+@datasets_option(help=unwrap("""An expression that fully or partially identifies dataset types that should
+                             constrain the yielded records. Only affects results when used with
+                             --collections."""))
+@collections_option(help=collections_option.help + " Only affects results when used with --datasets.")
+@where_option(help=whereHelp)
+@click.option("--no-check", is_flag=True,
+              help=unwrap("""Don't check the query before execution. By default the query is checked before it
```

Comment:
@TallJimbo what do we gain by having this option? When is disabling the check a good idea?

Reply:
The check can generate false positives for some valid-but-rare queries.  Those all involve _wanting_ multiple instruments _and_ a WHERE constraint a table derived from instrument (e.g. `exposure`) on a column in that table whose values are nevertheless instrument-agnostic (e.g. `exposure.exposure_time > 30`).  They then fall into two categories:
 - Cases where you do restrict the instruments (or skymaps) you want, but use `IN` instead of `=` and `OR`: `instrument IN ('HSC', 'DECam') AND exposure.exposure_time > 30`.  The checker just isn't smart enough to rewrite the `IN`.
 - Cases where you actually want to query all of the instruments (or skymaps) in the registry: just `exposure.exposure_time > 30`.  Note that I think we still want to check this case by default, because usually the user is thinking of a specific instrument but we don't know that.

Similar cases may exist for skymaps, but it's harder to think of practically useful cases given the attributes actually on the tract and patch tables.

I'm open to opinions on whether that's sufficiently niche as to make this option not worth its maintenance weight.